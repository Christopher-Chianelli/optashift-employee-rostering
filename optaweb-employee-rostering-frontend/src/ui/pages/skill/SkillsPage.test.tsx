/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { Skill } from 'domain/Skill';
import { getRouterProps } from 'util/BookmarkableTestUtils';
import { mockStore } from 'store/mockStore';
import { Map } from 'immutable';
import { mockRedux } from 'setupTests';
import { SkillsPage, SkillRow, EditableSkillRow } from './SkillsPage';

const noSkillsStore = mockStore({
  skillList: {
    isLoading: false,
    skillMapById: Map(),
  },
}).store;

const twoSkillsStore = mockStore({
  skillList: {
    isLoading: false,
    skillMapById: Map<number, Skill>()
      .set(0, {
        id: 0,
        version: 0,
        tenantId: 0,
        name: 'Skill 1',
      })
      .set(1,
        {
          id: 1,
          version: 0,
          tenantId: 0,
          name: 'Skill 2',
        }),
  },
}).store;

describe('Skills page', () => {
  it('should render correctly with no skills', () => {
    mockRedux(noSkillsStore);
    const skillsPage = shallow(<SkillsPage {...getRouterProps('/0/skill', {})} />);
    expect(toJson(skillsPage)).toMatchSnapshot();
  });

  it('should render correctly with a few skills', () => {
    mockRedux(twoSkillsStore);
    const skillsPage = shallow(
      <SkillsPage {...getRouterProps('/0/skill', {})} />,
    );
    expect(toJson(skillsPage)).toMatchSnapshot();
  });

  it('should render the viewer correctly', () => {
    const skill = { name: 'Skill', tenantId: 0, id: 1, version: 0 };
    mockRedux(twoSkillsStore);
    const viewer = shallow(<SkillRow {...skill} />);
    expect(toJson(viewer)).toMatchSnapshot();
  });

  it('should render the editor correctly', () => {
    const skill = { name: 'Skill', tenantId: 0, id: 1, version: 0 };
    mockRedux(twoSkillsStore);
    const editor = shallow(<EditableSkillRow skill={skill} isNew={false} onClose={jest.fn()} />);
    expect(toJson(editor)).toMatchSnapshot();
  });
});
