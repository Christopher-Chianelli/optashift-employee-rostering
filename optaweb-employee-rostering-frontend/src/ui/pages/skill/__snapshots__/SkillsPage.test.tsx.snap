// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills page should render correctly with a few skills 1`] = `
<TheTable
  columns={
    Array [
      Object {
        "name": "Trans(i18nKey=name)",
        "sorter": [Function],
      },
    ]
  }
  filterText=""
  filteredRows={
    Immutable.List [
      Object {
        "id": 0,
        "name": "Skill 1",
        "tenantId": 0,
        "version": 0,
      },
      Object {
        "id": 1,
        "name": "Skill 2",
        "tenantId": 0,
        "version": 0,
      },
    ]
  }
  history={
    Object {
      "action": "PUSH",
      "block": [MockFunction],
      "createHref": [MockFunction],
      "go": [MockFunction],
      "goBack": [MockFunction],
      "goForward": [MockFunction],
      "length": 1,
      "listen": [MockFunction],
      "location": Object {
        "hash": "",
        "pathname": "/0/skill",
        "search": "",
        "state": undefined,
      },
      "push": [MockFunction],
      "replace": [MockFunction],
    }
  }
  isReversed={false}
  itemsPerPage={10}
  location={
    Object {
      "hash": "",
      "pathname": "/0/skill",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/0/skill",
      "url": "localhost:8080/0/skill",
    }
  }
  newRowWrapper={[Function]}
  numOfFilteredRows={2}
  onSorterChange={[Function]}
  page={1}
  rowWrapper={[Function]}
  rowsInPage={
    Immutable.List [
      Object {
        "id": 0,
        "name": "Skill 1",
        "tenantId": 0,
        "version": 0,
      },
      Object {
        "id": 1,
        "name": "Skill 2",
        "tenantId": 0,
        "version": 0,
      },
    ]
  }
  sortByIndex={0}
  title="Trans(i18nKey=skills)"
/>
`;

exports[`Skills page should render correctly with no skills 1`] = `
<TheTable
  columns={
    Array [
      Object {
        "name": "Trans(i18nKey=name)",
        "sorter": [Function],
      },
    ]
  }
  filterText=""
  filteredRows={Immutable.List []}
  history={
    Object {
      "action": "PUSH",
      "block": [MockFunction],
      "createHref": [MockFunction],
      "go": [MockFunction],
      "goBack": [MockFunction],
      "goForward": [MockFunction],
      "length": 1,
      "listen": [MockFunction],
      "location": Object {
        "hash": "",
        "pathname": "/0/skill",
        "search": "",
        "state": undefined,
      },
      "push": [MockFunction],
      "replace": [MockFunction],
    }
  }
  isReversed={false}
  itemsPerPage={10}
  location={
    Object {
      "hash": "",
      "pathname": "/0/skill",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/0/skill",
      "url": "localhost:8080/0/skill",
    }
  }
  newRowWrapper={[Function]}
  numOfFilteredRows={0}
  onSorterChange={[Function]}
  page={1}
  rowWrapper={[Function]}
  rowsInPage={Immutable.List []}
  sortByIndex={0}
  title="Trans(i18nKey=skills)"
/>
`;

exports[`Skills page should render the editor correctly 1`] = `
<TableRow>
  <TableCell
    columnName="Trans(i18nKey=name)"
  >
    <TextInput
      onChange={[Function]}
      value="Skill"
    />
  </TableCell>
  <RowEditButtons
    isValid={true}
    onClose={[Function]}
    onSave={[Function]}
  />
</TableRow>
`;

exports[`Skills page should render the viewer correctly 1`] = `
<TableRow>
  <TableCell
    columnName="Trans(i18nKey=name)"
  >
    <Text>
      Skill
    </Text>
  </TableCell>
  <RowViewButtons
    onDelete={[Function]}
    onEdit={[Function]}
  />
</TableRow>
`;
